service/src/main.rs

```diff
@@ -881,16 +915,36 @@ fn enclave_account<E: EnclaveBase>(enclave_api: &E) -> AccountId32 {
 }
 
 /// Checks if we are the first validateer to register on the parentchain.
-fn we_are_primary_validateer(
+fn we_are_primary_worker(
 	node_api: &ParentchainApi,
-	register_enclave_xt_header: &Header,
+	shard: &ShardIdentifier,
+	enclave_account: &AccountId32,
 ) -> Result<bool, Error> {
-	let enclave_count_of_previous_block =
-		node_api.enclave_count(Some(*register_enclave_xt_header.parent_hash()))?;
-	trace!(
-		"enclave count is {} for previous block 0x{:?}",
-		enclave_count_of_previous_block,
-		register_enclave_xt_header.parent_hash()
-	);
-	Ok(enclave_count_of_previous_block == 0)
+	// are we registered? else fail.
+	node_api
+		.enclave(enclave_account, None)
+		.unwrap()
+		.expect("our enclave should be registered at this point");
+	trace!("our enclave is registered");
+	match node_api.primary_worker_for_shard(shard, None).unwrap() {
+		Some(enclave) =>
+			match enclave.instance_signer() {
+				AnySigner::Known(MultiSigner::Ed25519(primary)) =>
+					if primary.encode() == enclave_account.encode() {
```

Comment:
I don't think we need the encode here.

Reply:
well, the trait bound jungle annoyance....happy to hear an easy solution. this here would work, but it's an ugly abuse: `primary.0 == enclave_account.blake2_256()` as it compares `[u8; 32]`
I suggest to rather leave this for the next code scout to improve

Reply:
Alright